pkg/cmd/update/update.go

```diff
@@ -0,0 +1,286 @@
+		// go's time.Sub only works with time.Time, not time.Duration
+		previousTime = now.Add(-updateInterval)
+	}
+
+	timeSince := now.Sub(previousTime)
+	if timeSince >= updateInterval {
+		githubToken := ""
+		if cfg.IsPersisted() {
+			githubToken = cfg.GitHub.Token.Value
+		}
+
+		// Update the previousTime since we're checking
+		_, err := file.WriteString(now.Format(time.RFC3339))
+		if err != nil {
+			log.Warn().Err(err).Msgf("could not write timestamp to update status file: %s", tempFilePath)
+			return
+		}
+		err = file.Sync()
+		if err != nil {
+			log.Warn().Err(err).Msgf("could not sync update status file: %s", tempFilePath)
+			return
+		}
+
+		update(UpdateOpts{
+			githubToken:      githubToken,
+			skipConfirmation: false,
+			targetVersion:    "",
+			streams: userio.NewIOStreamsWithInteractive(
+				os.Stdin,
+				os.Stdout,
+				!nonInteractive,
+			),
+		})
+	} else {
+		timeLeft := (updateInterval - timeSince).Round(time.Second)
+		log.Debug().Msgf("next update check will be in %s", timeLeft)
+	}
+}
+
+func getCurrentExecutablePath() string {
+	execPath, err := os.Executable()
+	if err != nil {
+		log.Fatal().Err(err).Msg("Failed to get executable path")
+	}
+
+	absolutePath, err := filepath.Abs(execPath)
+	if err != nil {
+		log.Fatal().Err(err).Msg("Failed to get absolute executable path")
+	}
+
+	return absolutePath
+}
+
+func UpdateCmd(cfg *config.Config) *cobra.Command {
+	opts := UpdateOpts{
+		githubToken:   "",
+		targetVersion: "",
+	}
+	updateCmd := &cobra.Command{
+		Use:   "update",
+		Short: "Update corectl",
+		Long:  `Update to the latest corectl version.`,
+		Args:  cobra.MaximumNArgs(1),
+		Run: func(cmd *cobra.Command, args []string) {
+			if len(args) > 0 {
+				opts.targetVersion = args[0]
+			}
+			if cfg.IsPersisted() {
+				opts.githubToken = cfg.GitHub.Token.Value
+			}
+
+			nonInteractive, err := cmd.Flags().GetBool("non-interactive")
+			if err != nil {
+				log.Panic().Err(err).Msg("could not get non-interactive flag")
+			}
+
+			opts.streams = userio.NewIOStreamsWithInteractive(
+				os.Stdin,
+				os.Stdout,
+				!nonInteractive,
+			)
+
+			update(opts)
+		},
+	}
+
+	updateCmd.Flags().BoolVar(
+		&opts.skipConfirmation,
+		"skip-confirmation",
+		false,
+		"Auto approve confirmation prompt for update.",
+	)
+
+	return updateCmd
+}
+
+func update(opts UpdateOpts) {
+	if opts.targetVersion != "" {
+		log.Debug().Msgf("target version set to %s", opts.targetVersion)
+	}
+
+	wizard := opts.streams.Wizard("Checking for updates", "Retrieved version metadata")
+	githubClient := github.NewClient(nil)
+	if opts.githubToken != "" {
+		githubClient = githubClient.WithAuthToken(opts.githubToken)
+	}
+
+	var release *github.RepositoryRelease
+	var err error
+	if opts.targetVersion == "" {
+		release, err = git.GetLatestCorectlRelease(githubClient)
+	} else {
+		release, err = git.GetCorectlReleaseByTag(githubClient, opts.targetVersion)
+	}
+	if err != nil {
+		wizard.Abort(err)
```

Comment:
We shouldn't panic when the error is caused by any external means, we should handle it somehow or propagate to the caller